src/_app/use-cases/auth/refresh.ts

```diff
@@ -0,0 +1,52 @@
+import {
+  useAccessToken,
+  useRefreshToken,
+  useTokenValidity,
+} from '~/composables/auth/auth-cookies'
+import { PokeResponseStatus } from '~/_app/common/poke'
+import { createFoxyPokeClient } from '~/_app/infrastructure/foxy-poke-api'
+import { useSignOut } from './sign-out'
+
+export const useRefresh = () => {
+  const tokenValidity = useTokenValidity()
+  const accessToken = useAccessToken()
+  const refreshToken = useRefreshToken()
+
+  const signOut = useSignOut()
+
+  // creating separate client because it would call itself recursively
+  // TODO: solve this
+  const pokeApi = createFoxyPokeClient()
+
+  return async () => {
+    if (!accessToken.value && !refreshToken.value) {
+      return
+    }
+
+    if (tokenValidity.value) {
+      return
+    }
+
+    if (!accessToken.value || !refreshToken.value) {
```

Comment:
Create `exists` function with signature like exists<T>(target: T | undefined | null): target is T